### DISCUSSION TITLE
input data requirement

### DISCUSSION THREAD

**Discussion by taosheng-liu** [cvnlab/GLMsingle#92] [Q&A] [answered]
Hello,
  I recently found out this method and I've downloaded the code and the example works on my setup. Thanks for the great work! I have a couple of questions regarding the data input to the pipeline and I hope you could shed light on.
  1. In the demo the input data is a slice with all time points. I noticed that this image is apparently masked, with voxels outside the brain to be zeros. Is this necessary? My data are not masked so these peripheral voxels have a small positive value, instead of 0. Is this Ok, or should I mask it first?
  2. A related question is whether you recommend applying the pipeline to data from a ROI. I'm primarily conducting ROI-based analyses and it seems it would be straightforward to feed ROI time series to the pipeline. However, I wonder if the method needs information from the whole brain for some steps (e.g., denoise), hence whole-brain data is better for the input.
  
  Thank you for your help!!

Best,
Taosheng Liu

**Comment by kendrickkay** [comment, marked as answer]
1: masking is not necessary. You can provide unmasked data, and it should work fine.

2: The derivation of nuisnace regressors benefits from having access to lots of voxels (outside your ROI of interest). Hence, I would recommend giving the entire dataset to the GLMsingle tool.  That being said, GLMsingle should still run for the case of providing only a subset of your voxel time series.

**Reply by taosheng-liu** [reply]
Thanks for the quick reply. Much appreciated!
